app/errors.py

```diff
@@ -60,3 +65,31 @@ class MailSentFromReverseAlias(SLException):
 
 class ProtonPartnerNotSetUp(SLException):
     pass
+
+
+class ErrContactErrorUpgradeNeeded(SLException):
+    """raised when user cannot create a contact because the plan doesn't allow it"""
+
+    def error_for_user(self) -> str:
+        return f"Please upgrade to premium to create reverse-alias"
+
+
+class ErrAddressInvalid(SLException):
+    """raised when an address is invalid"""
+
+    def __init__(self, address: str):
+        self.address = address
+
+    def error_for_user(self) -> str:
+        return f"{self.address} is not a valid email address"
+
+
+class ErrContactAlreadyExists(SLException):
+    """raised when a contact already exists"""
+
+    # TODO: type-hint this as a contact when models are almost dataclasses and don't import errors
+    def __init__(self, contact):
```

Comment:
You can use `contact: "Contact"` as type annotation. pycharm is quite intelligent and can understand it.

Reply:
:+1: